pkg/query-service/app/traces/v3/query_builder.go

```diff
@@ -473,7 +473,16 @@ func addOffsetToQuery(query string, offset uint64) string {
 }
 
 func PrepareTracesQuery(start, end int64, queryType v3.QueryType, panelType v3.PanelType, mq *v3.BuilderQuery, keys map[string]v3.AttributeKey) (string, error) {
-	query, err := buildTracesQuery(start, end, mq.StepInterval, mq, constants.SIGNOZ_SPAN_INDEX_TABLENAME, keys, panelType)
+
+	stepInterval := mq.StepInterval
+	// update step interval for table view
+	if panelType == v3.PanelTypeTable {
+		stepInterval = (end - start)/1000
```

Comment:
This breaks with `toStartOfInterval`. Assume the start if 12:45 and the end is 1:15. When we use `toStartOfInterval(..., toIntervalSecond(...))`, you will have two points at 12:30 and 1:00 with data from 12:45-12:59:59 and 1:00-1:15 respectively when we should only return one value for the whole aggregation interval.